src/scores/stats/tests/diebold_mariano_impl.py

```diff
@@ -55,6 +55,9 @@ def diebold_mariano(
     Harvey, Leybourne and Newbold (1997) for why the t distribution may be preferred,
     especially for shorter timeseries.
 
+    If `da_timeseries` is a chunked array, data will be brought into memory during
+    this calculation due to the autocovariance implementation.
```

Comment:
Also, if you know the dimensions and labels of the output, based on the input size and labels, we can wrap this function so it is dask/lazy. It might still load everything into memory for the calculation, but will at least only do it when needed.

Reply:
Created a separate issue https://github.com/nci/scores/issues/59

I think it's lower priority since this test is generally run on smaller datasets. It would be nice to update it.